src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -657,6 +658,56 @@ private SupportedPlatform GetCurrentRunningOS()
             return context;
         }
 
+        internal static void UpdateLocationsAndMessageWithCurrentUri(IList<Location> locations, Message message, Uri updatedUri)
+        {
+            if (locations == null) { return; }
+
+            foreach (Location location in locations)
+            {
+                ArtifactLocation artifactLocation = location.PhysicalLocation?.ArtifactLocation;
+
+                if (message != null)
```

Comment:
>message [](start = 20, length = 7)

If `message` is `null` you can early return just as you do when `locations` is `null`.

Reply:
You can do it at the top, before entering the loop.

---
In reply to: [322888930](https://github.com/microsoft/sarif-sdk/pull/1670#discussion_r322888930) [](ancestors = 322888930)